docs/AQL/grammar/AqlParser.g4

```diff
@@ -0,0 +1,214 @@
+    ;
+
+
+selectExpr
+    : columnExpr (AS IDENTIFIER)?
+    ;
+
+fromExpr
+    : containsExpr
+    ;
+
+whereExpr
+    : NOT? identifiedExpr
+    | whereExpr AND whereExpr
+    | whereExpr OR whereExpr
+    | OPEN_PAR whereExpr CLOSE_PAR
+    ;
+
+orderByExpr
+    : identifiedPath order=(DESCENDING|DESC|ASCENDING|ASC)?
+    ;
+
+
+columnExpr
+    : identifiedPath
+    | aggregateFunctionCall
+    | functionCall
+    ;
+
+containsExpr
+    : classExprOperand (CONTAINS containsExpr)?
+    | containsExpr AND containsExpr
+    | containsExpr OR containsExpr
+    | OPEN_PAR containsExpr CLOSE_PAR
+    ;
+
+identifiedExpr
+    : EXISTS identifiedPath
+    | identifiedPath COMPARISON_OPERATOR identifiedOperand
+    | functionCall COMPARISON_OPERATOR identifiedOperand
+    | identifiedPath LIKE likeOperand
+    | identifiedPath MATCHES OPEN_CURLY matchesOperand CLOSE_CURLY
+    ;
+
+identifiedOperand
+    : primitive
+    | PARAMETER
+    | identifiedPath
+    | functionCall
+    ;
+
+identifiedPath
+    : IDENTIFIER pathPredicate? (SLASH objectPath)?
+    ;
+
+pathPredicate
+    : OPEN_BRACKET (standardPredicate | archetypePredicate | nodePredicate) CLOSE_BRACKET
+    ;
+
+standardPredicate
+    : objectPath COMPARISON_OPERATOR pathPredicateOperand
+    ;
+
+archetypePredicate
+    : ARCHETYPEID
+    | PARAMETER
+    ;
+
+nodePredicate
+    : NODEID (COMMA (STRING|PARAMETER))?
+    | ARCHETYPEID (COMMA (STRING|PARAMETER))?
+    | PARAMETER
+    | objectPath COMPARISON_OPERATOR pathPredicateOperand
+    | objectPath MATCHES REGEXPATTERN
+    | nodePredicate AND nodePredicate
+    | nodePredicate OR nodePredicate
+    ;
+
+pathPredicateOperand
+    : primitive
+    | objectPath
+    | PARAMETER
+    ;
+
+
+objectPath
+    : pathPart (SLASH pathPart)*
+    ;
+pathPart
+    : IDENTIFIER pathPredicate?
+    ;
+
+likeOperand
+    : STRING
+    | PARAMETER
+    ;
+matchesOperand
+    : valueListItem (COMMA valueListItem)*
+    | terminologyFunction
+    | URIVALUE
+    ;
+valueListItem
+    : primitive
+    | PARAMETER
+    | terminologyFunction
+    ;
+
+primitive
+    : STRING
+    | INTEGER
+    | FLOAT
+    | DOUBLE
+    | DATE
+    | BOOLEAN
+    ;
+
+
+
+classExprOperand
+    : IDENTIFIER IDENTIFIER? // RM_TYPE_NAME variable
+    | archetypedClassExpr
+    | versionedClassExpr
+    | versionClassExpr
+    ;
+
+// RM_TYPE_NAME [archetype_id]
+// RM_TYPE_NAME variable [archetype_id]
+archetypedClassExpr
+    : IDENTIFIER IDENTIFIER? archetypePredicate
+    ;
+versionedClassExpr
+    : VERSIONED_OBJECT IDENTIFIER? (OPEN_BRACKET standardPredicate OPEN_BRACKET)?
+    ;
+versionClassExpr
+    : VERSION IDENTIFIER? (OPEN_BRACKET (standardPredicate|versionPredicate) OPEN_BRACKET)?
+    ;
+versionPredicate
+    : LATEST_VERSION
+    | ALL_VERSIONS
+    ;
+
+functionCall
+    : terminologyFunction
+    | function
+    ;
+
+function
+    : IDENTIFIER OPEN_PAR functionArg (COMMA functionArg)* CLOSE_PAR
+    ;
+
+functionArg
```

Comment:
I think this is just a copy of what is called identifiedOperand in this grammar, but what i would call 'terminal' or similar.

Reply:
it is indeed - but I was not sure if it makes sense to use identifiedOperand as a function argument, (therefore i created a new (copy) rule.
should i just use identifiedOperand @wolandscat ?

Reply:
Yep, you don't want duplicate rules. Another illustration of why to name things as what they are, not where they appear ;)